pytket/tests/qir_test.py

```diff
@@ -151,34 +159,16 @@ def test_qir_from_pytket_circuit_and_pyqir_gateset(
     assert call_rz in data
 
 
-# def test_pyqir_builtin_gates_from_pytket_circuit() -> None:
-#     c = Circuit(2)
-#     c.CX(0, 1)
-#     mod_name = "Simple CX circuit"
-#     c_qir_str = circuit_to_qir_str(c, mod_name)
-#     print(c_qir_str)
-
-# def test_qir_str_measure() -> None:
-#     c = Circuit(1, 1)
-#     c.Measure(0, 0)
-#     module_name = "Simple Measure circuit"
-#     c_qir_str = circuit_to_qir_str(c, module_name)
-#     call = f"call %Result* @__quantum__qis__m__body(%Qubit* null)"
-#     assert call in c_qir_str
-
-
-# def test_qir_str_rz() -> None:
-#     c = Circuit(1)
-#     c.Rz(0.0, 0)
-#     module_name = "Simple Rz circuit"
-#     c_qir_str = circuit_to_qir_str(c, module_name)
-#     call = f"call void @__quantum__qis__rz__body(double 0.000000e+00, %Qubit* null)"
-#     assert call in c_qir_str
-
-
-# def test_extended_module() -> None:
-#     em = ExtendedModule('Extended Module', 2, 1)
-#     em.module.builder.call(em.zzmax, [em.module.qubits[0],em.module.qubits[1]])
-#     em_ir_str = em.module.ir()
-#     call = f"call void @__quantum__qis__zzmax__body(%Qubit* null, %Qubit* inttoptr (i64 1 to %Qubit*))"
-#     assert call in em_ir_str
+def test_bitwise_ops_for_quantinuum_gateset(
+    circuit_bitwise_ops: Circuit, bitwise_file: str
+) -> None:
+    with open(bitwise_file, "r") as input:
+        data = input.read()
+    print(data)
+    call_and = f"call void @__quantinuum__cis__and__body(%Result* %zero, %Result* %zero1, %Result* %zero2)"
+    call_or = f"call void @__quantinuum__cis__or__body(%Result* %zero3, %Result* %zero4, %Result* %zero5)"
+    call_xor = f"call void @__quantinuum__cis__xor__body(%Result* %zero6, %Result* %zero7, %Result* %zero8)"
```

Comment:
yeah I think we're going to have to make sure we have SSA calls with set outputs

Reply:
Sure. This is something I am not entirely clear with atm. Need to think through it a little.

Reply:
I guess, one thing that I can do here is to reflect better the DAG structure. If I understand correctly, the output of an `AND` operation should be an SSA variable of the same type. Then, there must be input/output bindings to be made at the start and end of each SSA block (using SORE/READ). Concretely, that would mean there should be some return type for the function calls.

Reply:
yes, though I think you only need one STORE/READ per Bit of the Circuit at the start and end of the dag, and just use SSA variables in the middle

Reply:
OK, it looks like the `add_external_function` only works for functions defined to return a `types.VOID`. I've tracked the Python binding and it finally calls an `emit_void_call` [here](https://github.com/qir-alliance/pyqir/blob/dd4f5fe56c8f8dc27483f96a40d63c572c37ce72/qirlib/src/generation/qir/instructions.rs#L183). There is also an `emit_call_with_return` but it only works for calls to `Measure`. Not sure there is a direct call to that method from Python. Will investigate.